openmmtools/tests/test_alchemy.py

```diff
@@ -2325,9 +2325,314 @@ def test_serialization(self):
             assert pickle.dumps(compound_state) == pickle.dumps(deserialized_state)
 
 
+def test_replace_variable_in_energy_string():
+    assert replace_variable_in_energy_string("r", "r", "s1") == "s1"
+    assert replace_variable_in_energy_string("a*r", "r", "s1") == "a*s1"
+    assert replace_variable_in_energy_string("a*r*r(2,1)", "r", "s1") == "a*s1*r(2,1)"
+    assert replace_variable_in_energy_string("r1","r","s1") == "r1"
+
+
+def test_is_variable_in_energy_string():
+    assert is_variable_in_energy_string("a*sigma*r; sigma=sigma1*sigma2", "sigma")
+    assert not is_variable_in_energy_string("a*sigma*r; sigma=sigma1*sigma2", "sigm")
+    assert is_variable_in_energy_string("r", "r")
+
+
+def test_is_nbfix_force():
+    charmm_solvated = testsystems.CharmmSolvated()
+    nbfix_forces = [charmm_solvated.system.getForce(i)
+                    for i in range(charmm_solvated.system.getNumForces())
+                    if is_nbfix_force(charmm_solvated.system.getForce(i))]
+    assert len(nbfix_forces) == 1
+
+
+def endstate_test(use_vswitch=False, pme_tol=1e-5, platform=openmm.Platform.getPlatformByName("Reference")):
+    """
+    Test that the energies in the end states of the alchemical calculation are consistent
+    with energies that do not use the alchemical forces.
+    """
+    testsystem = testsystems.CharmmSolvated(ewald_tolerance=pme_tol)
+    if use_vswitch:
+        forcefactories.use_custom_vdw_switching_function(testsystem.system, 1.0*unit.nanometer, 1.2*unit.nanometer)
+
+    # utility functions
+    def calc_energy(testsystem, platform=platform):
+        dummy_integrator = openmm.VerletIntegrator(1.0*unit.femtosecond)
```

Comment:
Could you implement this using the existing `test_alchemy.compute_energy` function in here?